### DISCUSSION TITLE
Align on fading/smoothing/forgetting terminology

### DISCUSSION THREAD

**Discussion by MaxHalford** [online-ml/river#1030] [Polls]
We're moving more and more towards models and algorithms that account for drift. This implies "forgetting" the past and focusing on recent data. Often the mechanism through which this happens is a smoothing factor, typically comprised in `[0, 1`]. We can see it in `EWMean`, `LossyCount`, `TextClust`, and at some point `BayesianLinearRegression` [will have it too](https://www.notion.so/maxhalford/Drift-resilient-Bayesian-linear-regression-4e9de8a8c766461ebf0010e91040f442).

This is great news, and it shows we're moving in the right direction of helping people deal with drift.

The one thing that irks me is that we're using different terminology in different places to refer to the same concept. `EWMean` calls this `alpha`, `LossyCount` calls this `forgetting`, `TextClust` calls this `fading`. There are probably other cases but I can't remember them off the top of my head.

I would like to suggest we use a single name for this kind of parameter. I've suggested a few options here, but feel free to suggest something else and to comment -- ping @online-ml/devs.

Once we agree I'll add this to our roadmap.

**Comment by raphaelsty** [comment]
What about `shift` or `memory`  or `drift` ? 🥸

Disadvantage is that `shift` could indicate an offset (as with the pandas library) and `memory` could rather make us think of RAM memory. 

Sorry but not sorry to add noise

**Reply by MaxHalford** [reply]
- `shift` yes it sounds like an offset
- `memory` that sounds like the `window_size` parameter. Maybe `memory` is a better term than `window_size`
- `drift` sounds like concept drift, which is the problem the parameter we're discussing is suppose to tackle. I would rather use a different name for the parameter

**Comment by Dennis1989** [comment]
I don't know of it is possible to really agree upon one term for that. We should keep in mind that there are different approaches when it comes to forgetting data. Different window modes can be for example used (landmark, sliding, damped). Using one fixed term may actually add confusion.

**Reply by smastelini** [reply]
I agree partially with you. The specific examples @MaxHalford  presented specifically concern exponentially fading weights. I think for those we can agree in a single term. But other cases should be handled separately.

**Reply by MaxHalford** [reply]
Yes I'm talking about fading windows, not regular windows.

**Comment by thorbjornwolf** [comment]
That sounds like a very good idea!

So, uh, let me also add to the noise from too many options 😄 
I've always liked `halflife` for exponential weights, but maybe that's because of the game.
[pandas's ewm has all of `com` (center of mass), `span`, `halflife` and `alpha`](https://pandas.pydata.org/pandas-docs/stable/reference/api/pandas.DataFrame.ewm.html), which is amazing but seems like too much to put in every relevant signature in `river`.

One wish: I hope the chosen name and its value correspond intuitively. For instance,
- an increase in `forgetting` should reduce the halflife
- an increase in `fading_factor` should reduce the halflife
- an increase in `smoothing` should increase the halflife
- an increase in `alpha` should... uhhh...

**Reply by MaxHalford** [reply]
Thanks that's helpful, especially the Halflife reference haha.

**Comment by MaxHalford** [comment]
Ok let's go with fading_factor! It makes sense the more I think about it. The factor part also makes it clear it's a parameter between 0 and 1. I'll make a pull request to make the necessary changes.

Thanks all!

**Reply by Dennis1989** [reply]
Sounds good. No work for me adjusting textclust ;)

**Reply by MaxHalford** [reply]
Haha right. About textClust, it's still on my todo to look into refactoring the code, I just haven't prioritized it yet :)
